diff --git a/glib/glib/giounix.c b/glib/glib/giounix.c
index 517e36b..794c4e5 100644
--- a/glib/glib/giounix.c
+++ b/glib/glib/giounix.c
@@ -187,8 +187,8 @@ g_io_unix_read (GIOChannel *channel,
   GIOUnixChannel *unix_channel = (GIOUnixChannel *)channel;
   gssize result;
 
-  if (count > SSIZE_MAX) /* At least according to the Debian manpage for read */
-    count = SSIZE_MAX;
+  if (count > LONG_MAX) /* At least according to the Debian manpage for read */
+    count = LONG_MAX;
 
  retry:
   result = read (unix_channel->fd, buf, count);